controllers/machinehealthcheck_controller.go

```diff
@@ -231,28 +234,29 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 		var message string
 
 		if m.Spec.UnhealthyRange == nil {
-			unhealthyLimitKey = "max unhealthy"
-			unhealthyLimitValue = m.Spec.MaxUnhealthy
+			logger.V(3).Info(
+				"Short-circuiting remediation",
+				totalTargetKeyLog, totalTargets,
+				maxUnhealthyKeyLog, m.Spec.MaxUnhealthy,
+				unhealthyTargetsKeyLog, len(unhealthy),
+			)
 			message = fmt.Sprintf("Remediation is not allowed, the number of not started or unhealthy machines exceeds maxUnhealthy (total: %v, unhealthy: %v, maxUnhealthy: %v)",
 				totalTargets,
 				len(unhealthy),
 				m.Spec.MaxUnhealthy)
 		} else {
-			unhealthyLimitKey = "unhealthy range"
-			unhealthyLimitValue = *m.Spec.UnhealthyRange
+			logger.V(3).Info(
+				"Short-circuiting remediation",
+				totalTargetKeyLog, totalTargets,
+				unhealthyRangeKeyLog, *m.Spec.UnhealthyRange,
+				unhealthyTargetsKeyLog, len(unhealthy),
+			)
 			message = fmt.Sprintf("Remediation is not allowed, the number of not started or unhealthy machines does not fall within the range (total: %v, unhealthy: %v, unhealthyRange: %v)",
 				totalTargets,
 				len(unhealthy),
 				*m.Spec.UnhealthyRange)
 		}
 
-		logger.V(3).Info(
```

Comment:
It seems like we could keep a single log call here, as well as below without needing to add an if/else block. The values can just be set in variables as they were, then only have one call out to the logger. The only thing missing from the existing code was to make sure the key was set to something `if remediationAllowed`.

Reply:
Thanks for feedback @stmcginnis. Is this suggesting just the opposite to https://github.com/kubernetes-sigs/cluster-api/pull/5110#discussion_r691291343? @vincepri 

Seems @joelspeed finds it more readable with the if/else https://github.com/kubernetes-sigs/cluster-api/pull/5110#pullrequestreview-733798949

I don't really have any strong opinion as far as we stop the controller from panicking.

@stmcginnis @JoelSpeed @vincepri please let me know if you want to merge it as it is or make any specific change.

Reply:
I don't have a really strong opinion on this, but having seen both, I think I prefer the way it is now. While it's more code, it's also more explicit about what's happening and I personally find it easier to follow this way.

Also, thinking about how the logging works, with this approach the log line will include the actual line numbers of the log calls and might make it easier to trace the logging and which if-statements were entered, as opposed with today where you have to work out the values that were substituted and then work out where they were set